docs/_releases/v2.0.0-pre.4/matchers.md

```diff
@@ -112,6 +112,26 @@ Requires the value to be a `Function`.
 Requires the value to be an `Array`.
 
 
+#### `sinon.match.array.deepEquals(iterable)`
+
+Requires an iterable value to be deep equal another one.
```

Comment:
Ah, jeez, this is harder. So we are either to choose between 
- expanding the namespace through `match.maps.*`, `match.sets.*`, etc

or 
- expanding the code under `match.iterable` for Maps, Sets, etc. 

I guess for the latter it would mean we should throw for all types we do not explicitly support (like for now: those missing a `length` property).

I need to get some input on this.